packages/sanity-studio/src/components/SearchArrayInput.tsx

```diff
@@ -0,0 +1,39 @@
+import { TextInput } from "@sanity/ui";
+import React from "react";
+import type { ArrayOfObjectsInputProps } from "sanity";
+
+export function SearchArrayInput({
+  members,
+  ...props
+}: ArrayOfObjectsInputProps): JSX.Element {
+  const [search, setSearch] = React.useState("");
+
+  const filteredMembers = !search
+    ? members
+    : members?.filter(
+        (member: any) =>
+          !member?.item?.value?.source ||
```

Comment:
This would only work if the array has items with the shape `{destination, source}`.
Can we somehow make it more general so it could work with any shape?
I wonder if JSON.stringify could be a good default to grab the value to filter by too 🤔

Reply:
JSON.stringify works perfectly => https://github.com/tinloof/sanity-kit/commit/a60c3fe85184c5e7708957c9a215d2ba5abdc647

Reply:
The value could be an object too and the default filter would include the keys of that object (e.g. "source", "destination").
Not really a problem, but I think it's worth adding a param to customize it (e.g. `itemToString(item: any)`)

Reply:
Just pushed a new [commit](https://github.com/tinloof/sanity-kit/pull/27/commits/f8ba14fdf5ae0aed43148b1ef02aae99009f9e60) that adds a `getAllValues` function to recursively get all values from an object.